presto-orc/src/main/java/com/facebook/presto/orc/stream/OrcInputStream.java

```diff
@@ -462,11 +467,19 @@ private void advance()
         if (isUncompressed) {
             buffer = ensureCapacity(buffer, chunkLength);
             length = compressedSliceInput.read(buffer, 0, chunkLength);
+            if (dwrfDecryptor.isPresent()) {
```

Comment:
This logic is correct for encrypted and compressed file (and a single chunk/page is uncompressed). We need to handle encrypted but uncompressed file.